explorer/explorerroutes.go

```diff
@@ -142,26 +142,80 @@ func (exp *explorerUI) DisapprovedBlocks(w http.ResponseWriter, r *http.Request)
 
 // NextHome is the page handler for the "/nexthome" path.
 func (exp *explorerUI) NextHome(w http.ResponseWriter, r *http.Request) {
+	// get top 30 blocks and trim each block to have just the fields required for this page
 	height := exp.blockData.GetHeight()
-
-	blocks := exp.blockData.GetExplorerFullBlocks(height, height-11)
-
+	blocks := exp.blockData.GetExplorerFullBlocks(height, height-30)
+	trimmedBlocks := make([]*TrimmedBlockInfo, 0, len(blocks))
+	for _, block := range blocks {
+		trimmedBlock := trimBlockInfo(block)
+		trimmedBlocks = append(trimmedBlocks, trimmedBlock)
+	}
 	exp.pageData.RLock()
 	exp.MempoolData.RLock()
 
+	// fetch votes for mempool
+	mempoolVotes := make([]*TxInfo, 0)
+	for _, tx := range exp.MempoolData.Votes {
+		if tx.VoteInfo.ForLastBlock == true {
+			exptx := exp.blockData.GetExplorerTx(tx.Hash)
+			mempoolVotes = append(mempoolVotes, exptx)
+		}
+	}
+
+	// fetch tickets for mempool
+	ticketsCount := len(exp.MempoolData.Tickets)
+	mempoolTickets := make([]*TxInfo, 0, ticketsCount)
+	for _, tx := range exp.MempoolData.Tickets {
+		exptx := exp.blockData.GetExplorerTx(tx.Hash)
+		mempoolTickets = append(mempoolTickets, exptx)
+	}
+
+	// fetch revocations for mempool
+	revCount := len(exp.MempoolData.Revocations)
+	mempoolRevs := make([]*TxInfo, 0, revCount)
+	for _, tx := range exp.MempoolData.Revocations {
+		exptx := exp.blockData.GetExplorerTx(tx.Hash)
+		mempoolRevs = append(mempoolRevs, exptx)
+	}
+
+	// fetch regular tx for mempool, and calculate total fees for mempool block
+	var mempoolFees float64 = 0
+	txCount := len(exp.MempoolData.Transactions)
+	mempoolTxs := make([]*TxInfo, 0, txCount)
+	for _, tx := range exp.MempoolData.Transactions {
+		exptx := exp.blockData.GetExplorerTx(tx.Hash)
+		if !exptx.Coinbase {
+			mempoolFees += exptx.Fee.ToCoin()
+			mempoolTxs = append(mempoolTxs, exptx)
+		}
+	}
+
+	// construct mempool object with properties required in template
+	mempoolData := MempoolData {
+		Subsidy:	  exp.pageData.HomeInfo.NBlockSubsidy,
+		Transactions: trimTxInfo(mempoolTxs),
+		Tickets:      trimTxInfo(mempoolTickets),
+		Votes:        trimTxInfo(mempoolVotes),
+		Revocations:  trimTxInfo(mempoolRevs),
+		Total:        exp.MempoolData.TotalOut,
+		Time:         exp.MempoolData.LastBlockTime,
+		Fees:		  mempoolFees,
+	}
+
```

Comment:
All of the new code in `NextHome` above is extremely inefficient.  I suggest to identify just what is missing in the `explorer.MempoolTx` type.  Do not use `GetExplorerTx` to get a `TxInfo` that is converted with `trimTxInfo`.  Instead only get the missing data.  If this requires a new function like `GetExplorerTxTrimmed` then you can add that, but I am **extremely** averse to running all these DB queries just for the home page.  Please have a careful look at how to be efficient about data handling and queries.  Before it was just the call to `GetExplorerFullBlocks`, and it is many times more demanding now.